fatiando/gravmag/transform.py

```diff
@@ -503,6 +507,105 @@ def derivz(x, y, data, shape, order=1, method='fft'):
     return deriv[padx: padx + nx, pady: pady + ny].ravel()
 
 
+def power_density_spectra(x, y, data, shape):
+    r"""
+    Calculates the Power Density Spectra of a 2D gridded potential field
+    through the FFT:
+
+    .. math::
+
+        \Phi_{\Delta T}(k_x, k_y) = | F\left{\Delta T \right}(k_x, k_y) |^2
+
+    .. note:: Requires gridded data.
+
+    .. note:: x, y, z and height should be in meters.
+
+    Parameters:
+
+    * x, y : 1D-arrays
+        The x and y coordinates of the grid points
+    * data : 1D-array
+        The potential field at the grid points
+    * shape : tuple = (nx, ny)
+        The shape of the grid
+
+    Returns:
+
+    * kx, ky : 2D-arrays
+        The wavenumbers of each Power Density Spectra point
+    * pds : 2D-array
+        The Power Density Spectra of the data
+    """
+    kx, ky = _fftfreqs(x, y, shape, shape)
+    pds = abs(numpy.fft.fft2(numpy.reshape(data, shape)))**2
+    return kx, ky, pds
+
+
+def radial_average(kx, ky, pds, max_radius=None, ring_width=None):
```

Comment:
This function is generic to any 2D array, so it might be better to rename the inputs to `x, y, data` instead of `kx, ky, pds` to indicate this. We can offer an example in the docstring to make it clear how to use this to calculate the power spectra.